src/cloudformation/newUserAndDbTemplate.yaml

```diff
@@ -0,0 +1,217 @@
+  pIasqlUrl:
+    Type: String
+    Description: Do not update
+  pA0Token:
+    Type: String
+    Description: Do not update
+
+Resources:
+
+  ###
+  ### IAM User
+  ###
+  rIamUser:
+    Type: AWS::IAM::User
+    Properties:
+      ManagedPolicyArns: 
+        - arn:aws:iam::aws:policy/AdministratorAccess
+      Tags: 
+        - Key: "iasql"
+          Value: "true"
+      UserName: !Sub ${AWS::StackName}-iasql-user-${pAwsRegion}
+
+  ###
+  ### IAM User Access Key
+  ###
+  rIamUserAccessKey:
+    Type: AWS::IAM::AccessKey
+    Properties:
+      UserName: !Ref 'rIamUser'
+
+  ####
+  #### Lambda Role
+  ####
+
+  rLambdaRole:
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName: !Sub ${AWS::StackName}-LambdaRole
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: "Allow"
+            Principal:
+              Service:
+                - "lambda.amazonaws.com"
+            Action:
+              - "sts:AssumeRole"
+      Policies:
+      - PolicyName: !Sub ${AWS::StackName}-LambdaPolicy
+        PolicyDocument:
+          Version: '2012-10-17'
+          Statement:
+            - Sid: CloudWatchLogsPermissions
+              Effect: Allow
+              Action:
+                - logs:CreateLogGroup
+                - logs:CreateLogStream
+                - logs:PutLogEvents
+              Resource: !Sub 'arn:${AWS::Partition}:logs:*:*:*'
+            - Sid: LambdaFunctionPermissionsforInvoke
+              Effect: Allow
+              Action:
+                - lambda:InvokeFunction
+                - lambda:InvokeAsync
+              Resource: "arn:aws:lambda:*:*:function:*"
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - lambda.amazonaws.com
+            Action:
+              - sts:AssumeRole
+
+  ####
+  #### Lambda Functions
+  ####
+
+  rLambdaFunction:
+    Type: AWS::Lambda::Function
+    DependsOn: rLambdaRole
+    Properties:
+      FunctionName: !Sub ${AWS::StackName}-LambdaFunction
+      Role: !GetAtt rLambdaRole.Arn
+      Environment:
+        Variables:
+          access_key_id: !Ref 'rIamUserAccessKey'
+          secret_access_key: !GetAtt 'rIamUserAccessKey.SecretAccessKey'
+          aws_region: !Ref 'pAwsRegion'
+          db_alias: !Ref 'pIasqlDbName'
+          a0_token: !Ref 'pA0Token'
+          api_url: !Ref 'pIasqlUrl'
+      Handler: index.handler
+      Timeout: 60
+      Runtime: nodejs14.x
+      Code:
+        S3Bucket: "slurp-cred" # TODO: make it an input?
+        S3Key: "newUserAndDb.zip" # TODO: to be updated
+      Tags:
+        -
+          Key: "Lambda Function Name"
+          Value: !Sub '${AWS::StackName}-LambdaFunction'
+
+  rLambdaFunctionCaller:
+    Type: AWS::Lambda::Function
```

Comment:
do we spin down the lambda fn after we are done?

Reply:
the lambda function runs and returns but it will stay listed as long as the stack is not deleted. Resources will only be deleted once the user deletes the CloudFormation stack.